### PR TITLE
Data model , proof and request

### PR DESCRIPTION
**Pull request by R11manish**
- data model
-  request 
- proof

with test cases

### CODE REVIEW

File: src/proof/proof.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+   * occurs, it throws an error.
+   * @param {createProof_mutationMutationVariables} [variables] - The `variables` parameter is an
+   * optional object that contains the variables needed for the `createProof_mutation` mutation. These
+   * variables can include any data that is required for the mutation to execute successfully, such as
+   * input values or authentication tokens.
+   * @returns The `createProof` function is returning the result of the `createProof_mutation` function
+   * call.
+   */
+
+  async createProof(inputVariables?: createProof_mutationMutationVariables) {
```
Comment (Siddharth9890) [inline src/proof/proof.ts:46]: it should not be optional i guess

File: test/data-model.test.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        required: ['name', 'age'],
+        additionalProperties: false,
+      },
+      description: 'Test description',
+      permissions: 'ALL',
+      title: 'Test',
+    });
+
+    expect(createDataModel?.title).toEqual('Test');
+  }, 40000);
```
Comment (Siddharth9890) [inline test/data-model.test.ts:38]: the timeout should be a variable which is constant

File: test/data-model.test.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  it('data model meta data', async () => {
+    const { dataModelsMetadata } = await api.dataModel.getDataModelMetaData();
+    expect(dataModelsMetadata.tags.length).toBeGreaterThanOrEqual(0);
+  });
+
+  it('find issuer by data model id', async () => {
+    const { issuersByDataModel } = await api.dataModel.getIssuersByDataModel(
+      process.env.DATAMODEL_ID as string,
+    );
+
```
Comment (Siddharth9890) [inline test/data-model.test.ts:81]: expect missing?

File: test/data-model.test.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  });
+
+  it('get total issuers by data model', async () => {
+    const { getTotalofIssuersByDataModel } =
+      await api.dataModel.getTotalofIssuersByDataModel(
+        process.env.DATAMODEL_ID as string,
+      );
+
```
Comment (Siddharth9890) [inline test/data-model.test.ts:89]: expect missing?

File: test/data-model.test.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  });
+
+  it('get issuers by data model count', async () => {
+    const { issuersByDataModelCount } =
+      await api.dataModel.getIssuersByDataModelCount(
+        process.env.DATAMODEL_ID as string,
+      );
```
Comment (Siddharth9890) [inline test/data-model.test.ts:96]: expect missing?

File: test/proof.test.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  api = new Gateway({
+    apiKey: apiKey,
+    token: token,
+  });
+});
+
+describe('proofs related test', () => {
+  it('create proof', async () => {
+    const {} = await api.proof.createProof({});
```
Comment (Siddharth9890) [inline test/proof.test.ts:22]: expect missing?

File: src/data-model/data-model.ts [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+   * `CreateDataModelInput`. It is an input object that contains the data needed to create a new data
+   * model.
+   * @returns a Promise that resolves to a value of type `createDataModel_mutationMutation`.
+   */
+  async createDataModel(createModelInput: CreateDataModelInput) {
+    try {
+      return await this.sdk.createDataModel_mutation({
+        input: createModelInput,
+      });
+    } catch (error: any) {
```
Comment (Siddharth9890) [inline src/data-model/data-model.ts:32]: remove :any part from all catch blocks just to maintain consistency

File: src/data-model/data-model.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+   * "organization" property is an optional input that represents the identification of an organization.
+   * It can be of type "InputMaybe<OrganizationIdentificationInput>", which means it can either be a
+   * valid organization identification input or undefined.
+   * @property {InputMaybe} user - The "user" property is of type "InputMaybe" which means it can either
+   * be a value of type "undefined" or a value of another type. The specific type of "user" is not
+   * provided in the code snippet, so it is unclear what type it should be.
+   * @returns The `getDataModelsCount` function is returning the result of the `dataModelsCount_query`
+   * method call, which is a promise.
+   */
+
```
Comment (Siddharth9890) [inline src/data-model/data-model.ts:101]: run pnpm format missing lines in test files as well

**Review by Siddharth9890 — CHANGES_REQUESTED** [review]

File: src/proof/proof.ts (hunk 1 above)
Comment (R11manish) [reply src/proof/proof.ts:46]: that is already optional , as u can see `?`

File: test/data-model.test.ts (hunk 2 above)
Comment (R11manish) [reply test/data-model.test.ts:38]: nopes , every test cases takes differenet amount of time , so we can't store into a variable.

File: test/data-model.test.ts (hunk 3 above)
Comment (R11manish) [reply test/data-model.test.ts:81]: lemme fix

File: src/data-model/data-model.ts (hunk 7 above)
Comment (R11manish) [reply src/data-model/data-model.ts:32]: cool

File: src/data-model/data-model.ts (hunk 8 above)
Comment (R11manish) [reply src/data-model/data-model.ts:101]: cool

**Pull request closed without merging**